shuffle-storage/src/main/java/com/alibaba/flink/shuffle/storage/partition/LocalFileMapPartitionFactory.java

```diff
@@ -327,6 +358,55 @@ public boolean useHddOnly() {
         return false;
     }
 
+    @Override
+    public String getStorageNameFromPath(String storagePath) {
+        try {
+            return Files.getFileStore(Paths.get(storagePath)).name();
+        } catch (IOException e) {
+            LOG.warn(
+                    String.format(
+                            "Failed to get the storage name of path %s, it may influence the "
+                                    + "effectiveness of storage usage restriction, which means that"
+                                    + " you may use more storage space than that configured by %s.",
+                            storagePath, StorageOptions.STORAGE_MAX_USABLE_SPACE_BYTES.key()),
+                    e);
```

Comment:
I print a log here because I am afraid that if we throw an exception and the user can not fix the for some reason, then the use can never start a cluster successfully. This is used only when the user want to limit the storage usage, I guess maybe a better way is to only throw exception when a user configured to limit the storage usage, if the user does not need to limit the storage usage, then we do not need to throw the exception. What do you think?

Reply:
ok, I also think this is a good choice.